src/Grid/grid_generators.jl

```diff
@@ -69,7 +69,7 @@ end
 
 function _generate_2d_nodes!(nodes, nx, ny, LL, LR, UR, UL)
       for i in 0:ny-1
-        ratio_bounds = i / (ny-1)
+        ratio_bounds = convert(eltype(LL), (i / (ny-1)))
```

Comment:
Can you leave a comment at the beginning of the function why this is done? It might not be obvious at first sight and someone might want to revert this.